core/src/utils/config.ts

```diff
@@ -124,6 +124,14 @@ export interface IonicConfig {
    */
   toastDuration?: number;
 
+  /**
+   * The selector that will be used to query the root of the Ionic application.
+   * This element is used for things like injecting overlay elements into the DOM and managing focus.
+   *
+   * @default 'ion-app'
+   */
+  appRootSelector?: string;
```

Comment:
For context, we needed to add this option because without `ion-app`, our overlay utils would default to injecting overlay element into the `body` of the DOM. However, this could cause problems in Frameworks like React where the actual React application was initialized inside a container in the DOM like `div id="root"`, so if the overlay were outside that container, things like event listeners would not work.

With this option, devs can specify where the overlay should get injected using any valid CSS selector